### PR TITLE
Dockerfile for the rust implementation

### PR DESCRIPTION
**Pull request by ojoggerst**
### Description

Changes the Dockerfile to work with the rust implementation.
This is a multi-stage dockerfile building in the first stage and the final stage there is only the executable.

Missing: build automation with github actions

Tested with:
- cargo test
- test scripts in tests
- examples

### CODE REVIEW

**Comment by reymondzzzz** [comment]
Hi! Thank for PR. I wanted to delete Dockerfile, because we deliver lsp server with plugins. What the reason of this PR?

**Comment by ojoggerst** [comment]
Hi @reymondzzzz,

The lsp server is also part of the refact docker image. The goal to run the lsp server in its own container. The webgui has a proxy for the lsp server and it is also started as part of the watch dog service or is the lsp server not needed anymore on the server side?

**Review by hazratisulton — APPROVED** [review]

**Review by olegklimov — APPROVED** [review]

File: src/global_context.rs [hunk 1]
```diff
@@ -39,6 +41,8 @@ pub struct CommandLine {
     pub basic_telemetry: bool,
     #[structopt(long, short="s", help="Send snippet telemetry (code snippets)")]
     pub snippet_telemetry: bool,
+    #[structopt(long, default_value="127.0.0.1", help="Binds by default to 127.0.0.1. This is compatible with having an HTTP server at the same time.")]
+    pub lsp_host: String,
```
Comment (olegklimov) [inline src/global_context.rs:45]: Hmm it's fine, but I mildly prefer to have just 1 parameter for host, just to reduce clutter (in code and in --help)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Stage 1: Builder Stage
```
Comment (olegklimov) [inline Dockerfile:1]: I think we need to explain why this exists. Something like "normally you don't need to run this in a docker, in fact when you use IDE plugins you get a binary file that just runs on the host without a container, but you might want to run this in a container for any reason that makes sense to you (such as isolation), and here's a Dockerfile to help you".

**Review by ojoggerst — COMMENTED** [review]
Fixed the suggested changes.

**Comment by JegernOUTT** [comment]
@ojoggerst can you resolve all conflicts? We're going to merge it ASAP after that

**Pull request closed without merging**